lib/langchain/llm/google_palm.rb

```diff
@@ -104,7 +109,7 @@ def chat(prompt: "", messages: [], context: "", examples: [], **options)
       response = client.generate_chat_message(**default_params)
       raise "GooglePalm API returned an error: #{response}" if response.dig("error")
 
-      response.dig("candidates", 0, "content")
+      Langchain::AIMessage.new(response.dig("candidates", 0, "content"))
```

Comment:
Why do we wrap low-level LLM API response into a higher-level `Message` class? I believe `AIMessage`/`Message` should be a part of `Conversation` higher-level abstraction. LLM classes should not return a `Message` instance.
cc @santib @andreibondarev

Reply:
that's a good point.. I did it here because I don't want to have a place like `Conversation` know how to access each LLM response keys, but maybe we can make use of another class so that neither `Conversation` nor `LLM::[x]` build the message? 🤔

At the same time, if we do that we'll go back to the original problem: the LLM responding just a string (which doesn't work fine with function calls)

Reply:
For reference, Python builds an `LLMResult` object https://github.com/langchain-ai/langchain/blob/3adb1e12cae290a4fe6500bed04e07219d652e74/libs/langchain/langchain/llms/openai.py#L876-L881

maybe that's what we need here

Reply:
Yeah, but they also separate the chat functionality into [langchain/chat_models/openai.py](https://github.com/langchain-ai/langchain/blob/3adb1e12cae290a4fe6500bed04e07219d652e74/libs/langchain/langchain/chat_models/openai.py).

In general I'd like to avoid introducing abstraction quickly because they're hard to get rid of afterwards.

So we know that we should change the LLM::* responses to return full hash responses, this has been mentioned by different people many times.

I agree with @alchaplinsky here I think the `LLM::OpenAI#chat()` should just return a hash and then we'd convert it to an AIMessage here: https://github.com/andreibondarev/langchainrb/blob/main/lib/langchain/conversation.rb#L62, just like it's done with the HumanMessage.

Reply:
but if we do that, we'll need to have `Conversation` know how to get the "content" and potentially other stuff (e.g. "function_call") from the different LLMs, right?

Reply:
Shouldn't the Conversation class be an encapsulation into complete functionality to have a conversation with an LLM? That's why I think the hash responses should be converted to Messages there. I think having Parsers or some sort of a hash lookup to remap is fine.

Reply:
I can take a stab at this refactor. I'd like to validate if this decoupling of `Message`s with `LLM`s apply only on the return value, or if it should also apply to the `prompt` (go back to `String` instead of a `Message`), and what happen with `messages` and `examples` which in the case of messages they are tailored for a conversation.

I'm fine changing the prompt + return value, but for messages and examples I think it'd be better to work with objects rather than hashes than need to conform to a format like `{ role: .., content: .. }`